test/core/event_engine/iomgr_event_engine/timer_list_test.cc

```diff
@@ -0,0 +1,246 @@
+/*
+ *
+ * Copyright 2015 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <string.h>
+
+#include <cstdint>
+#include <limits>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <grpc/grpc.h>
+
+#include "src/core/lib/event_engine/iomgr_engine/timer.h"
+#include "src/core/lib/gprpp/time.h"
+
+using testing::Mock;
+using testing::Return;
+using testing::StrictMock;
+
+namespace grpc_event_engine {
+namespace iomgr_engine {
+
+namespace {
+const int64_t kHoursIn25Days = 25 * 24;
+const grpc_core::Duration k25Days = grpc_core::Duration::Hours(kHoursIn25Days);
+
+class MockClosure : public experimental::EventEngine::Closure {
```

Comment:
I think this test might be better written using fakes instead of mocks.  It would require less boilerplate in the tests to set expectations, and it would avoid tying the tests so closely to the implementation (e.g., having to expect a particular number of calls to `Now()`).

Reply:
Leaving as is - I think having a test that verifies the number of calls to Now() is as expected is a feature here.